src/blocks/HeadingBlock/HeadingBlock.astro

```diff
@@ -0,0 +1,23 @@
+---
+import type { HeadingBlockFragment } from '@lib/datocms/types';
+
+export interface Props {
+  block: HeadingBlockFragment
+}
+
+const { block } = Astro.props;
```

Comment:
You can save yourself some typing by further destructure this:
```suggestion
const { block: { heading, subHeading } = Astro.props;
```

I do wonder if they should be called `heading` and `subHeading` though. The name `HeadingBlock` makes sense because heading is a role it fulfills. But the text inside the `HeadingBlock` is not a `heading`, I would argue that that property is the `title` of the `Heading`, and from there it would follow that the other value is `subtitle` (without camel case)